infertrade/algos/community/ta_regressions.py

```diff
@@ -34,11 +52,16 @@ def allocation_function(time_series_df: pd.DataFrame, *args, **kwargs) -> pd.Dat
             time_series_df = pipeline.fit_transform(time_series_df)
             return time_series_df
 
+	def create_allocation_function():
+            """Creates an allocation function."""
+            return deepcopy(allocation_function)
```

Comment:
How is deep copying a solution to this problem?

if the variable is not local to a function then
* It does not matter what value variable holds at the time of function creation because
* At the time of the function call, a variable may hold another value.

Reply:
Apologies - I misunderstood. I have amended the code. We could add a unit test to check the signals are now different?